src/server/actions/user-actions.ts

```diff
@@ -40,3 +40,55 @@ export const findOrCreateSiUserId = async (clerkId: string, name: string) => {
 
     return user.id
 }
+
+export const createUserAction = async (
+    clerkId: string,
+    name: string,
+    isResearcher: boolean,
+    memberId: string,
+    isReviewer: boolean,
+) => {
+    // Run all related inserts in a transaction for atomicity
+    return await db.transaction().execute(async (trx) => {
```

Comment:
There seems to be some overlap now with this method and the `findOrCreateSiUserId` method above. Maybe worth looking at to see if we need both? I can't tell 🤔